frontend/src/slices/articles/articles.slice.ts

```diff
@@ -54,13 +58,59 @@ const { reducer, actions, name } = createSlice({
       state.dataStatus = DataStatus.FULFILLED;
       state.article = action.payload;
     });
+    builder.addCase(reactToArticle.fulfilled, (state, action) => {
+      const { articleId, reaction: updatedReaction } = action.payload;
+
+      state.articles = state.articles.map((article) => {
+        if (article.id !== articleId) {
+          return article;
+        }
+
+        const reactionIndex = article.reactions.findIndex(
+          ({ id }) => id === updatedReaction.id,
+        );
+
+        if (reactionIndex === -1) {
+          article.reactions.push(updatedReaction);
+          return article;
+        }
+
+        article.reactions[reactionIndex] = updatedReaction;
+
+        return article;
+      });
+
+      state.articleReactionDataStatus = DataStatus.FULFILLED;
+    });
+    builder.addCase(deleteArticleReaction.fulfilled, (state, action) => {
+      const { articleId, reactionId } = action.payload;
+
+      state.articles = state.articles.map((article) => {
+        if (article.id === articleId) {
+          const reactionIndex = article.reactions.findIndex(
+            ({ id }) => id === reactionId,
+          );
+
+          article.reactions.splice(reactionIndex, 1);
```

Comment:
```suggestion
          const reactionsToUpdate = article.reactions.slice();
          
          reactionsToUpdate.splice(reactionIndex, 1);

          return {
            ...article,
            reactions: reactionsToUpdate
          };
```

Reply:
I thought that we can change value directly